File: src/main/java/br/gov/pf/model/entity/AbstractEntity.java
package br.gov.pf.model.entity;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

/**
 * Created by sartori on 13/07/17.
 */

@MappedSuperclass
public abstract class AbstractEntity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "created_at")
    private Date createdAt;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "updated_at")
    private Date updatedAt;

    public Long getId() {
        return id;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    @PrePersist
    public void setCreatedAt() {
        this.createdAt = new Date();
        this.updatedAt = new Date();
    }

    @PreUpdate
    public void setUpdatedAt() {
        this.updatedAt = new Date();
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }


}

File: src/main/java/br/gov/pf/model/entity/AbstractEntity.kt
package br.gov.pf.model.entity;

import java.io.Serializable
import java.util.*
import javax.persistence.*

/**
 * Created by sartori on 13/07/17.
 */

@MappedSuperclass
abstract class AbstractEntity : Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long = 0L

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "created_at")
    var createdAt: Date = Date()

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "updated_at")
    var updatedAt: Date = Date()


    @PrePersist
    fun perPersist() {
        this.createdAt = Date()
        this.updatedAt = Date()
    }

    @PreUpdate
    fun preUpdate() {
        this.updatedAt = Date()
    }
}